### DISCUSSION TITLE
Should I use redis cache adapter for Result cache & Metadata Cache for better performance?

### DISCUSSION THREAD

**Discussion by Axedyson** [mikro-orm/mikro-orm#2278] [Q&A]
Since I'm using Redis in my project I thought I might as well connect the [Result Cache](https://mikro-orm.io/docs/caching) and the [Metadata Cache](https://mikro-orm.io/docs/next/metadata-cache#providing-custom-cache-adapter) to my redis instance. But I would like to know if it has any benefits or if it's not needed. I guess the answer would be "it depends". For context, I'm planning on having the redis instance in the same server as the main node server, maybe I'll separate the two into their own dedicated servers in the future for some reason.

For the actual adapter I'm thinking this one: https://github.com/ramiel/mikro-orm-cache-adapter-redis

**Comment by B4nan** [comment]
For metadata cache it does not matter much, you definitely can, but it makes more sense to do it for result cache, as you could have one cache for multiple app instances.

**Reply by Axedyson** [reply]
Okay I've tried to use this [redis cache adapter](https://github.com/ramiel/mikro-orm-cache-adapter-redis) for the resultCache only, but for some reason when I try to use the cli e.g. `yarn mikro-orm cache:clear` even though the message: "Metadata cache was successfully cleared" appears, it never actually cleared the cache, it just hangs forever without any progress:

<img width="393" alt="Skærmbillede 2021-10-17 kl  16 09 28" src="https://user-images.githubusercontent.com/28491857/137630982-8b018e7e-4dc3-4106-b7be-6cbf4bd83a53.png">

This is my config file:
<img width="665" alt="Skærmbillede 2021-10-17 kl  16 11 28" src="https://user-images.githubusercontent.com/28491857/137631063-52bb90b8-f912-451e-8160-250ecc516256.png">
The expiration setting is just for testing purposes, I will set it back to 1000 ms.

**Reply by Axedyson** [reply]
Also I just spend a couple of hours trying to figure out why nothing got cached, but by reading the source code I finally found out that all you have to do is pass `{ cache: true }` to the `options` parameter to a query (e.g. `find(...)`) before it gets cached. I think this page: https://mikro-orm.io/docs/caching/ could've made that a lot clearer. The page states the following:
_MikroORM has simple result caching mechanism._
_By ***default***, in memory cache is used, that is shared for the whole MikroORM instance. ***Default*** expiration is 1 second_. 

I thought that caching was enabled by default, but now I know you have to explicitly set an option for it to work. Since it's not enabled by default it must mean that most db responses doesn't need to be cached. Is there some best practice of when to cache db responses that you could share? Because otherwise I think I'm just gonna drop the caching functionality
